equinix/resource_fabric_cloud_router.go

```diff
@@ -243,6 +272,17 @@ func projectCloudRouterTerraformToGo(projectTerraform []interface{}) fabricv4.Pr
 
 	return project
 }
+func marketplaceSubscriptionCloudRouterTerraformToGo(marketplaceSubscriptionTerraform []interface{}) fabricv4.MarketplaceSubscription {
+	if len(marketplaceSubscriptionTerraform) == 0 {
+		return fabricv4.MarketplaceSubscription{}
+	}
+	marketplaceSubscription := fabricv4.MarketplaceSubscription{}
+	marketplaceSubscriptionMap := marketplaceSubscriptionTerraform[0].(map[string]interface{})
+	subscriptionUUID := marketplaceSubscriptionMap["uuid"].(string)
```

Comment:
You added type as Optional/Computed but you aren't accounting for type as an input here. Additionally, do empty string checks before setting, and use setters instead of direct assignment on attributes.